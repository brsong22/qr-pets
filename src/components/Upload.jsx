import React from 'react';
import axios from 'axios';
import Button from '@material-ui/core/Button';
import Input from '@material-ui/core/Input';

class Upload extends React.Component {
  constructor() {
    super();

    this.state = {
      fileList: [],
    };
    this.handleClick = this.handleClick.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange({ target }) {
    this.setState({
      fileList: target.files,
    });
  }

  async handleClick() {
    const { fileList } = this.state;
    const file = fileList[0];
    const { name, type } = file;
    const { data } = await axios.post('/upload', { name, type });
    const uploadResponse = await axios.put(data, file);
  }

  render() {
    const { fileList } = this.state;
    const containsNoFile = fileList.length === 0;

    return (
      <div className="upload">
        <Input
          onChange={this.handleChange}
          type="file"
          id="file-chooser"
          disableUnderline
        />
        <Button
          color="primary"
          disabled={containsNoFile}
          onClick={this.handleClick}
          variant="contained"
        >
          {'Upload Picture'}
        </Button>
      </div>
    );
  }
}

export default Upload;
