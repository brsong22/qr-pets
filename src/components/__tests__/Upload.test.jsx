import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Input from '@material-ui/core/Input';
import Upload from '../Upload';
import axios from 'axios';

jest.mock('axios');

describe('Upload', () => {
  const firstPicture = { name: 'frodo', type: 'html/jpeg' };
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Upload />);
  });

  it('renders as expected', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('stores file from input in state', () => {
    const InputComponent = wrapper.find(Input);
    const event = {
      target: {
        files: [firstPicture],
      },
    };

    InputComponent.simulate('change', event);

    expect(wrapper.state().fileList).toEqual([firstPicture]);
  });
});
