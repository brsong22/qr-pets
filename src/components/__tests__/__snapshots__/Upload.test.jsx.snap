// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload renders as expected 1`] = `
<div
  className="upload"
>
  <WithStyles(Input)
    disableUnderline={true}
    id="file-chooser"
    onChange={[Function]}
    type="file"
  />
  <WithStyles(Button)
    color="primary"
    disabled={true}
    onClick={[Function]}
    variant="contained"
  >
    Upload Picture
  </WithStyles(Button)>
</div>
`;
